syntax = "proto2";

package NCloud.NBlockStore;

option go_package = "github.com/ydb-platform/nbs/cloud/blockstore/tools/testing/eternal_tests/eternal-load/lib/config";

message TRangeConfig {
  optional uint64 StartOffset = 1;
  optional uint64 RequestBlockCount = 2;
  optional uint64 RequestCount = 3;
  optional uint64 Step = 4;
  optional uint64 StartBlockIdx = 5;
  optional uint64 LastBlockIdx = 6;
  optional uint64 NumberToWrite = 7;
  optional uint64 WriteParts = 8;
}

message TTestConfig {
  optional uint64 TestId = 1;
  required string FilePath = 2;
  required uint32 IoDepth = 3;
  required uint64 FileSize = 4;
  required uint64 BlockSize = 5;
  required uint32 WriteRate = 6;
  optional uint64 RangeBlockCount = 7;
  repeated TRangeConfig Ranges = 8;
  required string SlowRequestThreshold = 9 [default = "10s"];
  // Used for a testcases, where write rate is periodically replaced
  // with read rate and vice versa.
  optional string AlternatingPhase = 10;
  // Used for tests only.
  optional uint64 MaxWriteRequestCount = 11;
}
